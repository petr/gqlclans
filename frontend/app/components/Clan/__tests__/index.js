import React from 'react'
import { Clan, mutation } from '../'

import { shallow } from 'enzyme'

const setUp = (props) => {
    const defaultProps = {
        clan: {
            clanId: '123',
            name: 'Gold LordS',
            tag: 'GLS',
            color: '#1010E6',
            members: [
                { 
                    name: 'Zibiro', 
                    accountId: '30915272', 
                    role: 'executive_officer', 
                    __typename: 'Member' 
                },
            ],
            messages: [],
            __typename: 'Clan',
        },
        mutate: jest.fn(() => Promise.resolve()),
    }

    return {
        wrapper: shallow(<Clan {...defaultProps} {...props} />),
        props: {...defaultProps, ...props},
    }
}

describe('Clan specification', () => {
    it('render query shape', () => {
        expect(mutation).toMatchSnapshot()
    })

    it('render render clan data', () => {
        const { wrapper, props } = setUp()
        const columns = wrapper.find('TableRowColumn')
        expect(columns.at(0).prop('children').join('')).toEqual(`[${props.clan.tag}]-${props.clan.name}`)
        expect(columns.at(1).prop('children')).toEqual(props.clan.members.length)
        expect(columns.at(2).prop('children').props.style.backgroundColor).toEqual(props.clan.color)
    })

    it('mutate clan message', () => {
        const { wrapper, props } = setUp()
        wrapper.setState({ message: 'Fake message' })
        const messageButton = wrapper.find('RaisedButton')
        messageButton.simulate('click')

        expect(props.mutate.mock.calls.length).toEqual(1)
        expect(props.mutate).toHaveBeenCalledWith({
            variables: {
                body: 'Fake message',
                clanId: '123',
            }
        })
    })
})