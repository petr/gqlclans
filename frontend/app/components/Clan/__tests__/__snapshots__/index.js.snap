// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clan specification render query shape 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "addMessage",
      },
      "operation": "mutation",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "body",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "body",
                  },
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "clanId",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "clanId",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "addMessage",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "message",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "body",
                        },
                        "selectionSet": null,
                      },
                    ],
                  },
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "success",
                  },
                  "selectionSet": null,
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "String",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "body",
            },
          },
        },
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "ID",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "clanId",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 198,
    "start": 0,
  },
}
`;
